### PR TITLE
Small change to bio-diesel recipe

### PR DESCRIPTION
**Pull request by Gilferedir**
#### Summary
```SUMMARY: Content "A more realistic bio-diesel recipe"```

#### Purpose of change
The game currently contains a number of items that have no current use. A number of these items are various tools and chemicals pertaining to chemistry.  This change will provide a use for separation funnels and a means to produce a small amount of glycerol much like one would see in a real life laboratory setting. This will hopefully be the first of many changes to help expand on the in-game chemistry options. 

#### Describe the solution
Addition of the requirement of a separation funnel for bio-diesel production.

Addition of the production of glycerol as a byproduct of bio-diesel production.

#### Describe alternatives you've considered
Leaving things as is, as the current bio-diesel recipe is still very viable.

Creating some uses for glycerol before allowing its production. 

Possibly reducing the crafting time of bio-diesel to accommodate the need of a separation funnel.

#### Additional context
Bio-diesel production at the lab scale is a fairly simple process and can be found in most college chemistry curriculum. An example can be found here: [https://www.luc.edu/media/lucedu/sustainability-new/pdfs/biodiesel/Biodiesel%20Curricula%20-%20Version%205.0.pdf](url)

### CODE REVIEW

**Comment by TechyBen** [comment]
Is the separation funnel craftable?

**Comment by Gilferedir** [comment]
> Is the separation funnel craftable?

Not currently, though it might be a good addition to the art of glassblowing book. Right now I believe it is found randomly in labs and research facilities.

File: data/json/recipes/chem/fuel.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
   },
   {
     "result": "biodiesel",
+    "byproducts": [ [ "chem_glycerol" ] ],
```
Comment (I-am-Erk) [inline data/json/recipes/chem/fuel.json]: Did you check that the quantity of glycerol produced here is appropriate to the proportions in the recipe?

**Comment by Gilferedir** [comment]
> Hallelujah, finally separation quality is seeing use in game.
> 
> As long as the amount of glycerol produced is correct I see no problems here.

That was something I still wasn't sure about and might need to be adjusted. The ratio by mass should be 1 tonne biodiesel to 100kg glycerol. There doesn't seem to be a mass for glycerol so I'm not sure how those things should work when that is the case. Currently 1 batch makes 1L bio-diesel and 0.01L glycerol by volume. The density of glycerol is ~1.26 mL/g and bio-diesel ~0.88 mL/g. I don't know how accurate these things should be or if the game will allow such specific quantities. If a volume ratio is acceptable, perhaps 10 units of glycerol should be produced per batch. I'm not quite sure how to do that though. should it be `"byproducts": [ [ "chem_glycerol", 10 ] ],` or `"byproducts": [ [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], [ "chem_glycerol" ], ],` as each unit of glycerol is 0.01L?

**Comment by Gilferedir** [comment]
I'm going to hold of on making any changes for now until I have a better understanding of how things work and feel more confident about the changes I'm imposing on others.

**Pull request closed without merging**